Tests/SwiftDocCTests/Infrastructure/ExternalReferenceResolverTests.swift

```diff
@@ -1155,4 +1155,232 @@ class ExternalReferenceResolverTests: XCTestCase {
             "doc://com.external.testbundle/externally/resolved/path",
         ])
     }
+
+    func assertMarkupHasResolvedLink(markupsWithResolvedLink: [String], actualMarkup: [Markup]) throws {
+        let value: String = try XCTUnwrap(actualMarkup.first?.format().trimmingCharacters(in: .whitespaces))
+        XCTAssert(
+            markupsWithResolvedLink.contains(value),
+            "Markup does not have resolved link\nActual markdown: \(value)\nExpected markdown options:\n\(markupsWithResolvedLink.joined(separator: "\n"))"
+        )
+    }
+
+    func withExampleDocumentation(_ files: [any File], path: String, block: (Symbol) throws -> Void) throws {
```

Comment:
The `any` keyword here new in Swift 5.8 so we can't use it yet. 

I've been working to make it _possible_ for us to update to Swift 5.9 but we haven't actually make the update yet. I plan to do that after March 15 when the `release/6.0` branch will be cut from `main`. If this is planned to be merged before then or be cherry-picked into the 6.0 release then it should remove the `any` keyword here.

Reply:
Also, I get the `with...` + closure pattern here but I feel that it mostly results in extra indentation for the caller:

```swift
withExampleDocumentation(
   file1, file2, file3
) { symbol in
   XCTAssertEqual(symbol.something, ...)
}
```

compared to 
```swift
let symbol = withExampleDocumentation(
   file1, file2, file3
)
XCTAssertEqual(symbol.something, ...)

```

Reply:
This style with a closure is something I'm used to writing Ruby code. Ruby loves to make this sort of code slightly more human readable, at the cost of some indentation.

Anyway I've rewritten the unit tests anyway, and in the new version I didn't use a closure - the helper functions just return values directly.

Reply:
I also removed the use of `any` here in the test, but we'll need to merge #861 first since I added a use of `some` in the production code in ExternalReferenceWalker.